IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/wolfssl_demo.c

```diff
@@ -294,78 +336,129 @@ static void Tls_client()
     /* attempt to connect TLS server */
 
     PeerAddr.sin_addr = FreeRTOS_inet_addr(TLSSERVER_IP);
-    PeerAddr.sin_port = FreeRTOS_htons(TLSSERVER_PORT);
+    PeerAddr.sin_port = FreeRTOS_htons(p->port);
 
     ret = FreeRTOS_connect(socket, &PeerAddr, sizeof(PeerAddr));
 
     if (ret != 0) {
-        printf("ERROR FreeRTOS_connect: %d\n",ret);
+        msg(pcName, p->id, "ERROR FreeRTOS_connect: %d\n",ret);
     }
 
+
+
+   #if defined(TLS_MULTITHREAD_TEST)
+    msg(pcName, p->id, " Ready to connect.\n");
+    xStatus = xSemaphoreTake(p->xBinarySemaphore, portMAX_DELAY);
+    if (xStatus != pdTRUE) {
+        msg(pcName, p->id, " Error : Failed to xSemaphoreTake\n");
+        goto out;
+    }
+   #endif
+
     /* create WOLFSSL object */
     if (ret == 0) {
         ssl = wolfSSL_new(ctx);
         if (ssl == NULL) {
-            printf("ERROR wolfSSL_new: %d\n", wolfSSL_get_error(ssl, 0));
+            msg(pcName, p->id, "ERROR wolfSSL_new: %d\n", 
+                                        wolfSSL_get_error(ssl, 0));
             ret = -1;
         }
     }
+
     if (ret == 0) {
         #ifdef WOLFSSL_RENESAS_TSIP_TLS
-        tsip_set_callback_ctx(ssl, &userContext);
+         #if !defined(TLS_MULTITHREAD_TEST)
+          memset(&userContext, 0, sizeof(TsipUserCtx));
+          tsip_set_callback_ctx(ssl, &userContext);
+         #else
+          if (p->port - TLSSERVER_PORT == 0) {
+             memset(&userContext_taskA, 0, sizeof(TsipUserCtx));
+             userContext_taskA.devId = 1;
```

Comment:
userContext_taskA.devId will be cleared to zero in tsip_set_callback_ctx, since it calls ForceZero to initialize the passed userContext_taskA. If it is the user's responsibility to initialize the userContext, remove ForceZero from the function. And is `devId`  appropriate as the member name for this purpose? How about contextId?

Reply:
Please look at `wc_CryptoCb_CryptInitRenesasCmn()`. I have changed my mind to assign the id automatically rather by user. The id is increased by each wc_CryptoCb_CryptInitRenesasCmn() call internally.

Since the id is mainly Crypt Callback use, the name is still valid, I think.